packages/react/src/context/use-context-mutator.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useContext, useRef } from 'react';
```

Comment:
This file was moved into a new dir.

The only changes are additional js-docs and I changed the `setGlobal` option to `default` to keep in line with our other terminology (we call this the default context not the global context when we are talking about domains)